### PR TITLE
Add Tau4 keyboard, with default and i2cscanner keymaps

### PR DESCRIPTION
**Pull request by ogaskell**
## Description

I added the necessary config files to a new directory in keyboards/ for the keyboard I recently designed and built. I am planning on selling this keyboard sometime in the future, hence I am pushing my config and keymaps to the qmk source. It includes the default keymap, and an i2cscanner keymap from [https://git.nixfifty.com/nix/qmk_firmware/-/blob/805f5cb72bb04c10327b40ed0c41e3848ad9ca75/keyboards/handwired/onekey/keymaps/i2c_scanner/keymap.c](here) incase this is needed to setup the OLED or EEPROM.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/tau4/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        break;
+    default: //  for any other layers, or the default layer
+        setrgb (0x00,  0x00, 0x00, (LED_TYPE *)&led[7]);
+        break;
+    }
+
+    rgblight_set();
+
+    return update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);
+}
```
Comment (drashna) [inline keyboards/tau4/keymaps/default/keymap.c:84]: The rgb light layers feature does the same thing, but "better". 
https://docs.qmk.fm/#/feature_rgblight?id=lighting-layers

The advantage to this is that it's set in the core code. Specifically, after animations are sent, but before the LED array is sent to the strip.  This prevents flickering issues, and is more performant, in theory. 

```suggestion
const rgblight_segment_t PROGMEM my_raise_layer[] = RGBLIGHT_LAYER_SEGMENTS(
    {7, 1, 1, 202, 110}
);
const rgblight_segment_t PROGMEM my_lower_layer[] = RGBLIGHT_LAYER_SEGMENTS(
    {7, 1, 125, 188, 80}
); 
const rgblight_segment_t PROGMEM my_adjust_layer[] = RGBLIGHT_LAYER_SEGMENTS(
    {7, 1, 32, 25, 127}
);
const rgblight_segment_t PROGMEM my_numpad_layer[] = RGBLIGHT_LAYER_SEGMENTS(
    {7, 1, HSV_BLUE}
);
const rgblight_segment_t PROGMEM my_default_layer[] = RGBLIGHT_LAYER_SEGMENTS(
    {7, 1, HSV_OFF}
);

const rgblight_segment_t* const PROGMEM my_rgb_layers[] = RGBLIGHT_LAYERS_LIST(
    my_default_layer,
    my_raise_layer,
    my_lower_layer,
    my_adjust_layer,
    my_numpad_layer
);

void keyboard_post_init_user(void) {
    rgblight_layers = my_rgb_layers;
}

layer_state_t layer_state_set_user(layer_state_t state) {
    state = update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);

    rgblight_set_layer_state(_QWERTY, layer_state_cmp(state, _QWERTY));
    rgblight_set_layer_state(_RAISE, layer_state_cmp(state, _RAISE));
    rgblight_set_layer_state(_LOWER, layer_state_cmp(state, _LOWER));
    rgblight_set_layer_state(_ADJUST, layer_state_cmp(state, _ADJUST));
    rgblight_set_layer_state(_NUMPAD, layer_state_cmp(state, _NUMPAD));

    return state;
}
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/tau4/keymaps/default/keymap.c (hunk 1 above)
Comment (ogaskell) [reply keyboards/tau4/keymaps/default/keymap.c]: I've made commits for all these changes, and it seems to build and work fine on my keyboard. 

The lighting layers is a much better solution, it's now working with animations so thanks for that :+1: .

Comment (drashna) [reply keyboards/tau4/keymaps/default/keymap.c]: I'm glad to hear it, and you are very welcome!

**Review by drashna — APPROVED** [review]

File: keyboards/tau4/readme.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+τ.4
+===
```
Comment (fauxpark) [inline keyboards/tau4/readme.md:2]:
```suggestion
# τ.4
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/tau4/rules.mk [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+ENCODER_ENABLE = yes        # Rotary Encoder support
+OLED_DRIVER_ENABLE = yes    # OLED display support
+# EEPROM_DRIVER ?= i2c        # Driver for external EEPROM chip
+# This is currently not working due to QMK not officially supporting the chip used on the Tau4, I am working on a fix.
```
Comment (noroadsleft) [inline keyboards/tau4/rules.mk:26]: Enables the Community Layouts feature.

```suggestion
# This is currently not working due to QMK not officially supporting the chip used on the Tau4, I am working on a fix.

LAYOUTS = ortho_4x12
```

File: keyboards/tau4/readme.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+A 40% ortholinear keyboard, developed by Oliver Gaskell. More info on [gaskell.tech](https://gaskell.tech/tau/four.html)
+
+* Keyboard Maintainer: [Oliver Gaskell](https://github.com/ogaskell)
+* Hardware Supported: Tau4 PCB rev1.0, Tau4 PCB rev1.1
+* Hardware Availability: Group Buy - Opens 05 Feb 2021 [More Details](https://gaskell.tech/tau/four.html)
+
+Default keymap:
+
+![Keymap](http://gaskell.tech/tau/images/tau4keymap.png)
```
Comment (noroadsleft) [inline keyboards/tau4/readme.md:13]: This would be better in `keyboards/tau4/keymaps/default/readme.md`.

```suggestion
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]
GitHub tip: You can apply multiple suggestions to a single commit by using the [Files Changed tab](https://github.com/qmk/qmk_firmware/pull/11361/files).

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**